lib/kredis/types/slots.rb

```diff
@@ -19,18 +19,23 @@ def reserve
           release
         end
       else
-        if incr <= available
+        if available?
+          incr
           true
         else
-          release
           false
         end
       end
     end
   end
 
   def release
-    decr
+    if get.to_i.between?(1, available)
```

Comment:
Why not just `get.to_i > 0`?

Reply:
Ohh yes. I missed to recognise after adding `available?` check in `reserve`. I will update it. Thank you.